src/AutoRest.CSharp/Common/AutoRest/Plugins/GeneratedCodeWorkspace.cs

```diff
@@ -120,7 +167,15 @@ public static async Task<GeneratedCodeWorkspace> Create(string projectDirectory,
                 }
             }
 
-            return new GeneratedCodeWorkspace(generatedCodeProject);
+            var compilation = (await generatedCodeProject.GetCompilationAsync())!;
```

Comment:
Building multiple compilations is expensive. Lets move this logic into GetGeneratedFilesAsync right before the values are needed.

Reply:
Fixed